### ISSUE TITLE
`object-does-not-match-filename` lints prints incorrect filename

### ISSUE THREAD

**Issue by maxonfjvipon** [objectionary/lints#296]
If `/program/@name` is `foo.x.main` and `/program/objects/o[1]/@name` is `main` and `/program/metas/meta[head='package']/tail` is `f`, which is:
```
+package f

[] > main
```
Then I see in log: `Object "main" does not match with filename ""`.

So the `/program/@name` is not printed correctly if it does not have the same prefix as `+package` alias

**Comment by h1alexbel** [comment]
@maxonfjvipon assign me, please

**Comment by h1alexbel** [comment]
@maxonfjvipon in this case, I think `main` (object name) should be equal to `main` (from program name). However, in the following example (with the same program name: `foo.x.main`):

```eo
+package f

[] > foo
```

we should report the warning, since `foo` does not match with `main`. Am I correct?

**Comment by maxonfjvipon** [comment]
@h1alexbel when the program name is `foo.x.main` it means that the object must have parent `x`, which must have parent `foo`. That all means that object `main` must be in package `foo.x`. That all again means that there must be a file `foo/x/main.xmir`. But in my example we have object `main` in package `f`. So it fails correctly, but it does not print program name correctly

**Comment by h1alexbel** [comment]
@maxonfjvipon I think we should compare only the filename in this lint, so in `foo.x.main`, `main` is the filename. In the reported case we should not fail, since object name `main` equals to the filename `main`. WDYT?

**Comment by maxonfjvipon** [comment]
@h1alexbel [this](https://github.com/objectionary/lints/issues/110) is the original ticket and yes, there's no any word about package.
Then we should decide if this lint should compare 
1. full `/program/@name` with `(package + object name)` or just 
2. last part of `/program/@name` (for example `main` from `foo.x.main`) with just `object name`

@yegor256 WDYT?

**Comment by yegor256** [comment]
@maxonfjvipon I believe, the `@name` in the `/program` must be equal to package + object name

**Comment by maxonfjvipon** [comment]
@h1alexbel see above, let's fix it

**Comment by h1alexbel** [comment]
@yegor256 should we make a new release?

**Comment by yegor256** [comment]
@rultor release, tag is `0.0.37`

**Comment by h1alexbel** [comment]
@maxonfjvipon the problem should be fixed, try again with new version, please

**Comment by h1alexbel** [comment]
@maxonfjvipon still an issue?

**Comment by maxonfjvipon** [comment]
@h1alexbel no, thanks
